src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/Constants.cs

```diff
@@ -23,14 +24,37 @@ static Constants()
 
             Debug.Assert(assemblyVersion != null);
 
+
+            var majorVersion = typeof(Constants).Assembly.GetName().Version.Major;
+            var minorVersion = typeof(Constants).Assembly.GetName().Version.Minor;
+            var os = getOperatingSystem();
+            var runtime = ".NET";
+            var runtimeVersion = RuntimeInformation.FrameworkDescription;
+
             // assembly version attribute should always be present
             // but in case it isn't then don't include version in user-agent
             if (assemblyVersion != null)
             {
-                userAgent += "/" + assemblyVersion.InformationalVersion;
+                userAgent = $"Microsoft SignalR/{majorVersion}.{minorVersion} ({assemblyVersion.InformationalVersion}; {os}; {runtime}; {runtimeVersion})";
```

Comment:
Why are you using the AssemblyInformationalVersionAttribute for the InformationalVersion, but not the major and minor versions?

Reply:
That's how we get the detailed version of the package